x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table_filters/tags_filter_popover.tsx

```diff
@@ -60,75 +34,78 @@ const TagsFilterPopoverComponent = ({
+    changedOption: EuiSelectableOption
+  ) => {
+    setSelectableOptions(newOptions);
+    toggleSelectedGroup(changedOption.label, selectedTags, onSelectedTagsChanged);
+  };
 
   useEffect(() => {
-    setFilterTags(
-      sortedTags.filter((tag) => tag.toLowerCase().includes(searchInput.toLowerCase()))
-    );
-  }, [sortedTags, searchInput]);
+    const selectedTagsSet = new Set(selectedTags);
+    const newSelectableOptions: EuiSelectableOption[] = sortedTags.map((label) => ({
+      label,
+      checked: selectedTagsSet.has(label) ? 'on' : undefined,
+    }));
+
+    setSelectableOptions(newSelectableOptions);
+  }, [sortedTags, selectedTags]);
+
+  const triggerButton = (
+    <EuiFilterButton
+      grow
+      iconType="arrowDown"
+      onClick={() => setIsTagPopoverOpen(!isTagPopoverOpen)}
+      numFilters={tags.length}
+      isSelected={isTagPopoverOpen}
+      hasActiveFilters={selectedTags.length > 0}
+      numActiveFilters={selectedTags.length}
+      data-test-subj="tags-filter-popover-button"
+    >
+      {i18n.TAGS}
+    </EuiFilterButton>
+  );
 
   return (
     <EuiPopover
       ownFocus
-      button={
-        <EuiFilterButton
-          grow={true}
-          data-test-subj={'tags-filter-popover-button'}
-          iconType="arrowDown"
-          onClick={() => setIsTagPopoverOpen(!isTagPopoverOpen)}
-          numFilters={tags.length}
-          isSelected={isTagPopoverOpen}
-          hasActiveFilters={selectedTags.length > 0}
-          numActiveFilters={selectedTags.length}
-        >
-          {i18n.TAGS}
-        </EuiFilterButton>
-      }
+      button={triggerButton}
       isOpen={isTagPopoverOpen}
       closePopover={() => setIsTagPopoverOpen(!isTagPopoverOpen)}
       panelPaddingSize="none"
       repositionOnScroll
+      panelProps={{
+        'data-test-subj': 'tags-filter-popover',
+      }}
     >
-      <PopoverContentWrapper>
-        <EuiPopoverTitle>
-          <EuiFieldSearch
-            placeholder="Search tags"
-            value={searchInput}
-            onChange={onSearchInputChange}
-            isClearable
-            aria-label="Rules tag search"
-          />
-        </EuiPopoverTitle>
-        <ScrollableDiv>{tagsComponent}</ScrollableDiv>
-        {filterTags.length === 0 && (
-          <EuiFlexGroup gutterSize="m" justifyContent="spaceAround">
-            <EuiFlexItem grow={true}>
-              <EuiPanel>
-                <EuiText>{i18n.NO_TAGS_AVAILABLE}</EuiText>
-              </EuiPanel>
-            </EuiFlexItem>
-          </EuiFlexGroup>
+      <EuiSelectable
+        searchable
+        searchProps={{
+          placeholder: 'Search tags',
+        }}
+        aria-label="Rules tag search"
+        options={selectableOptions}
+        onChange={handleSelectableOptionsChange}
+        emptyMessage={i18n.NO_TAGS_AVAILABLE}
+        noMatchesMessage={i18n.NO_TAGS_AVAILABLE}
+      >
+        {(list, search) => (
+          <div style={{ width: 275 }}>
```

Comment:
let's a variable for  `275`  https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx#magic-numbersstrings

Reply:
Good note